src/ZPublisher/HTTPRequest.py

```diff
@@ -1644,18 +1644,19 @@ class TemporaryFileWrapper(_TemporaryFileWrapper):
     and doesn't complain about such a move either.
     """
 
-    unlink = staticmethod(unlink)
-    isfile = staticmethod(isfile)
+    if PY2:
+        unlink = staticmethod(unlink)
+        isfile = staticmethod(isfile)
```

Comment:
But why do they need to be defined on the class at all?  The `__del__` method could just call them:

```python
        def __del__(self):
            self.close()
            if os.path.isfile(self.name):
                os.unlink(self.name)
```

At first I thought it was so that they could be mocked out more easily, but that doesn't appear to be the case.